mm2src/lp_ordermatch.rs

```diff
@@ -4069,72 +4098,57 @@ async fn cancel_previous_maker_orders(
+            // This checks that the order hasn't been removed by another process
+            if removed_order_mutex.is_some() {
+                maker_order_cancelled_p2p_notify(ctx.clone(), &order).await;
+                delete_my_maker_order(ctx.clone(), order.clone(), MakerOrderCancellationReason::Cancelled)
+                    .compat()
+                    .await
+                    .ok();
+            }
         }
-
-        // return the order to [`OrdermatchContext::my_maker_orders`]
-        my_actual_maker_orders.insert(uuid, order);
     }
-
-    *my_maker_orders = my_actual_maker_orders;
 }
 
 pub async fn update_maker_order(ctx: &MmArc, req: MakerOrderUpdateReq) -> Result<MakerOrder, String> {
     let ordermatch_ctx = try_s!(OrdermatchContext::from_ctx(ctx));
-    let my_maker_orders = ordermatch_ctx.my_maker_orders.lock().await;
-
-    let (base_coin, rel_coin, original_price, original_volume, updated_conf_settings, matches, reserved_amount) =
+    let order_mutex = {
+        let my_maker_orders = ordermatch_ctx.my_maker_orders.lock();
         match my_maker_orders.get(&req.uuid) {
-            Some(order) => {
-                if order.has_ongoing_matches() {
-                    return ERR!("Can't update an order that has ongoing matches");
-                }
-                let base = order.base.as_str();
-                let base_coin: MmCoinEnum = match try_s!(lp_coinfind(ctx, base).await) {
-                    Some(coin) => coin,
-                    None => return ERR!("Base coin {} has been removed from config", base),
-                };
-
-                let rel = order.rel.as_str();
-                let rel_coin: MmCoinEnum = match try_s!(lp_coinfind(ctx, rel).await) {
-                    Some(coin) => coin,
-                    None => return ERR!("Rel coin {} has been removed from config", rel),
-                };
+            Some(order) => order.clone(),
+            None => return ERR!("There is no order with UUID {}", req.uuid),
+        }
+    };
 
-                let original_conf_settings = order.conf_settings.unwrap();
+    let mut order = order_mutex.lock().await;
+    if order.has_ongoing_matches() {
+        return ERR!("Can't update an order that has ongoing matches");
+    }
 
-                let updated_conf_settings = OrderConfirmationsSettings {
-                    base_confs: req.base_confs.unwrap_or(original_conf_settings.base_confs),
-                    base_nota: req.base_nota.unwrap_or(original_conf_settings.base_nota),
-                    rel_confs: req.rel_confs.unwrap_or(original_conf_settings.rel_confs),
-                    rel_nota: req.rel_nota.unwrap_or(original_conf_settings.rel_nota),
-                };
+    let base = order.base.as_str();
+    let rel = order.rel.as_str();
+    let (base_coin, rel_coin) = match find_pair(ctx, base, rel).await {
+        Ok(Some(c)) => c,
+        _ => return ERR!("Base coin {} or rel coin {} has been removed from config", base, rel),
```

Comment:
"Has been removed from config" is not a correct statement here. The coins might be still present in the config but not activated yet.